File: codegen/results/v002/qwen3.5_4b-nvfp4_think/20260412_085910/typescript/002_word_frequency/solution.ts
import { readFileSync } from 'fs';

const content = readFileSync('input/text.txt', 'utf8');

// Split into words, strip punctuation (keep only letters), lowercase
const words = content
  .match(/([a-z0-9]+)/g) ?? []
  .map(word => word.toLowerCase());

if (!words.length) {
  process.exit(0);
}

const freqMap: Map<string, number> = new Map();

for (const word of words) {
  const count = freqMap.get(word) ?? 0;
  freqMap.set(word, count + 1);
}

// Convert to array and sort by count descending, then alphabetically ascending
const freqArray = Array.from(freqMap.entries());
freqArray.sort((a, b) => {
  if (b[1] !== a[1]) {
    return b[1] - a[1];
  }
  return a[0].localeCompare(b[0]);
});

for (const [word, count] of freqArray) {
  console.log(`${word}: ${count}`);
}

File: codegen/results/v002/qwen3.5_4b-nvfp4_think/20260412_085910/python/002_word_frequency/solution.py
import re
from collections import Counter

# Read input file
with open('input/text.txt', 'r') as f:
    text = f.read()

# Extract words: lowercase + alphabetic characters only
words = [word for word in text.lower()]
words = [re.sub(r'[^a-z]', '', char) for char in words]

# Group into actual words (by matching sequences between non-letters and spaces/newlines etc.)
import re
words = re.findall(r'[a-z]+', text.lower())

# Count word frequencies
word_counts = Counter(words)

# Sort by count descending, then alphabetically ascending for ties
sorted_words = sorted(word_counts.items(), key=lambda x: (-x[1], x[0]))

# Print results in required format
for word, count in sorted_words:
    print(f'{word}: {count}')